vm/src/stdlib/pystruct.rs

```diff
@@ -88,10 +88,29 @@ where
     I: Sized + Iterator<Item = char>,
 {
     let mut codes = vec![];
+    let mut use_repeat = false;
```

Comment:
Idea:

Instead of two mutable variables, rewrite the loop as following:

```rust
while chars.peek().is_some() {
    // determine repeat operator:
    let repeat = match chars.peek() {
        '0' .. '9' => {
           let mut repeat = 0;
           while let Some('0' .. '9') = chars.peek()  {
               let current_digit = c.to_digit(10).unwrap();
               repeat = repeat * 10 + current_digit;
           }
           Some(repeat)
        },
        _ => None,
    };

    // determine format char:
    let c = chars.next();
    match c {
      Some('b' | 'B' | ... ) => {
          if let Some(repeat) = repeat {
             // emit format char repeated
          } else {
             // emit format char normally
          }
      },
      _ => return illegal format code
   }
}
```

Reply:
Nice approach. I applied this code to the PR.